src/utils/saml.ts

```diff
@@ -156,17 +156,26 @@ export const getPreValidateResponse =
         Error,
         { readonly InResponseTo: NonEmptyString; readonly Response: Element }
       > = pipe(
-        responsesCollection,
-        E.fromPredicate(
-          (coll) => coll.length < 2,
-          (_) => new Error("SAML Response must have only one Response element")
+        responsesCollection.item(0),
+        // this check is bound to the next one, because we can receive no Response based on the official validator guidelines
+        E.fromNullable(
+          new Error("Missing Response element inside SAML Response")
         ),
-        E.map((coll) => coll.item(0)),
-        // this check is bound to the previous one, because we can receive no Response based on the official validator guidelines
-        E.chain(
-          E.fromNullable(
-            new Error("Missing Response element inside SAML Response")
-          )
+        E.chainFirst(
+          // updated versions of xmldom will convert any additional root node in a text node(https://github.com/advisories/GHSA-crh6-fp67-6883)
+          // to ensure if more than one samlp:Response node was provided, we must check if the sibling node is present, returning an error afterwards
+          (item) =>
+            pipe(
+              item.nextSibling,
```

Comment:
maybe using `fromPredicate` we can avoid to swap and make this more readable?

Reply:
thank you for this suggestion, I applied it in https://github.com/pagopa/io-spid-commons/pull/147/commits/960fbbf1e0e15cd3c3f440aa648ff3227edd2f4e